source/common/http/conn_manager_impl.cc

```diff
@@ -913,6 +931,18 @@ void ConnectionManagerImpl::ActiveStream::maybeEndDecode(bool end_stream) {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::disarmRequestTimer() {
+  if (request_timer_) {
+    request_timer_->disableTimer();
+  }
+}
+
+void ConnectionManagerImpl::ActiveStream::maybeDisarmRequestTimer() {
+  if (request_info_.lastUpstreamTxByteSent().has_value()) {
```

Comment:
I'm not keen on peppering ConnectionManagerImpl with calls to this check, especially since I think some of these calls may happen well after last byte is actually sent. I think you'd have to add a new method on StreamDecoderFilterCallbacks to get rid of it, however.

Reply:
It might be worth asking for opinions in the dev slack channel, but it seems cleaner to me to trigger this behavior directly rather than inferring it based on the state of the RequestInfo.

Reply:
As in add a callback that disables the timer? Or add a callback that modifies a variable that tracks whether the last byte has been sent?